### PR TITLE
Error on invalid jib.httpTimeout values at plugin front-ends

### LINKED ISSUE #582: Pushing to registry is timing out
**Description of the issue**:
Our docker-registry is sometimes a little bit slow, so the building-process inside our CI-pipeline fails with the message `Build image failed: Read timed out`. The timeout i selected inside my settings.xml for this server is not respected.

**Expected behavior**:
Respecting the Timeout inside the servers-tag in settings.xml

**Steps to reproduce**:
Try to push an image to an slow docker-registry. Then set the timeout for this registry inside your settings.xml and test ist again. You will see that the timeout is not respected.

Tested with version 0.9.6

### PR DESCRIPTION
**Pull request by chanseokoh**
Fixes #582.

### CODE REVIEW

**Comment by chanseokoh** [comment]
You can start reviewing. Once #656 is merged, I'll change the target base for merging.

**Comment by chanseokoh** [comment]
Hmm... I just saw the `.jib.frontend` package in jib-core. I guess the code should go in there.

**Comment by chanseokoh** [comment]
Also, looks like I should throw `MojoExecutionException` for Maven and `GradleException` for Gradle.

File: jib-core/src/main/java/com/google/cloud/tools/jib/frontend/SystemPropertyValidator.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+   * @param exceptionFactory factory to create an exception with the given description
+   * @param <T> the exception type to throw if invalid values
+   * @throws T if invalid values
+   */
+  public static <T extends Throwable> void checkHttpTimeoutProperty(
+      Function<String, T> exceptionFactory) throws T {
+    try {
+      String value = System.getProperty("jib.httpTimeout");
+      if (value != null && Integer.parseInt(value) < 0) {
+        throw exceptionFactory.apply("negative value of jib.httpTimeout");
```
Comment (coollog) [inline jib-core/src/main/java/com/google/cloud/tools/jib/frontend/SystemPropertyValidator.java]: How about a wording of "jib.httpTimeout cannot be negative: -1234" (with the negative value)?

File: jib-core/src/main/java/com/google/cloud/tools/jib/frontend/SystemPropertyValidator.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      }
+    } catch (NumberFormatException ex) {
+      throw exceptionFactory.apply("non-integer value of jib.httpTimeout");
```
Comment (coollog) [inline jib-core/src/main/java/com/google/cloud/tools/jib/frontend/SystemPropertyValidator.java]: How about a wording of "jib.httpTimeout must be an integer: notaninteger" (with the attempted value)?

File: jib-core/src/main/java/com/google/cloud/tools/jib/frontend/SystemPropertyValidator.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    }
```
Comment (coollog) [inline jib-core/src/main/java/com/google/cloud/tools/jib/frontend/SystemPropertyValidator.java]: Let's include a private constructor for this static utility class.

**Comment by coollog** [comment]
Cools, we'll be able to do a v0.9.7 release after this is merged.

**Comment by TadCordle** [comment]
@coollog should we get the last bit of https://github.com/GoogleContainerTools/jib/issues/586 in as well?

**Comment by coollog** [comment]
@TadCordle We could probably do the `equals(registry) || startsWith(registry + "/")` since its quick and easy.

**Comment by chanseokoh** [comment]
> @TadCordle We could probably do the `equals(registry) || startsWith(registry + "/")` since its quick and easy.

To clarify, [this is being handled correctly](https://github.com/GoogleContainerTools/jib/blob/master/jib-core/src/main/java/com/google/cloud/tools/jib/registry/credentials/json/DockerConfigTemplate.java#L147) after #650.

**Pull request merged**